### PR TITLE
feat: Enable `UpgradePlugin`s to restrict to a specific graph and update the knora base version to 26

### PR DESCRIPTION
**Pull request by seakayone**
### Task Description/Number

This PR allows an `UpgradePlugin` to confine its changes to a specific set of graphs. This speed up the upgrade process substantially. If `UpgradePlugins` only need to update the builtIn ontology graphs there is no need to drop all other ontology or data graphs. 

Moves `knora-ontologies` to `webapi/src/main/resources/knora-ontologies` in order to make the ontologies part of the classpath. This is necessary for local development in order to run the `UpgradePlugins` locally. Hence, we can remove adding them to the container manually. Retains a symbolic link on the root level of this repo so that other scripts and test code still can find them. Cleaning this up is left for [a followup issue ](https://linear.app/dasch/issue/DEV-3236/dsp-api-remove-symbolic-link-for-knora-ontologies) in order to keep the size of this PR manageable.

Replaces `NoopPlugin` with a more adequately named `MigrateOnlyBuiltInGraphs`. This plugin is updating the _builtIn_ graphs (i.e. the ones found in `knora-ontologies`) once applied.

Upgrades the knora base version to `26` and applies above migration plugin in order to update the _builtIn_ graphs to the latest version.

### PR Type

- [ ] build/chore: maintenance tasks (no production code change)
- [ ] docs: documentation changes (no production code change)
- [x] feat: represents new features
- [ ] fix: represents bug fixes
- [ ] perf: performance improvements
- [ ] refactor: represents production code refactoring
- [ ] test: adding or refactoring tests (no production code change)

### Basic requirements for bug fixes and features

- [ ] Tests for the changes have been added
- [ ] Docs have been added / updated

### Does this PR introduce a breaking change?

- [ ] Yes

### Does this PR change client-test-data?

- [ ] Yes

### CODE REVIEW

**Review by mpro7 — APPROVED** [review]
Thanks for this update, it looks promising! Although I approve it please wait in this case for all reviewers feedback.

File: webapi/src/main/scala/org/knora/webapi/store/triplestore/impl/TriplestoreServiceLive.scala [hunk 1]
```diff
@@ -431,10 +430,18 @@ case class TriplestoreServiceLive(
    * @param outputFile           the output file.
    * @return a string containing the contents of the graph in N-Quads format.
    */
-  override def downloadRepository(outputFile: Path): Task[Unit] = {
-    val request = new HttpGet(paths.repository)
-    request.addHeader("Accept", mimeTypeApplicationNQuads)
-    doHttpRequest(request, writeResponseFileAsPlainContent(outputFile)).unit
+  override def downloadRepository(outputFile: Path, graphs: GraphsForMigration): Task[Unit] = {
```
[the remark this answers is not in the document]
Comment (seakayone) [reply webapi/src/main/scala/org/knora/webapi/store/triplestore/impl/TriplestoreServiceLive.scala:433]: see 431969b4dc16b4dbc3548f059dbbc611cdaff6ac

**Comment by seakayone** [comment]
> Thanks for this update, it looks promising! Although I approve it please wait in this case for all reviewers feedback.

I appreciate your review, have you been able to test this out locally?

**Comment by mpro7** [comment]
> > Thanks for this update, it looks promising! Although I approve it please wait in this case for all reviewers feedback.
> 
> I appreciate your review, have you been able to test this out locally?

Yes, repository upgrade test passed locally.

**Comment by BalduinLandolt** [comment]
> Moves `knora-ontologies` to `webapi/src/main/resources/knora-ontologies` ...

@jnussbaum FYI - my guess is that symlinks will not work over github, so it's probably best to change it in TOOLS right away when this is merged. (And RDU will then have to update their DSP-TOOLS version, I suppose.)

**Review by BalduinLandolt — APPROVED** [review]
As discussed, I like the changes and I'm looking forward to continue improving the plugin situation, but this takes away the immediate pressure, which is great!

**Comment by BalduinLandolt** [comment]
> > > Moves `knora-ontologies` to `webapi/src/main/resources/knora-ontologies` ...
> > 
> > 
> > @jnussbaum FYI - my guess is that symlinks will not work over github, so it's probably best to change it in TOOLS right away when this is merged. (And RDU will then have to update their DSP-TOOLS version, I suppose.)
> 
> @BalduinLandolt So if I get you right, we have to modify our URLs as follows:
> 
> ```
> webapi/scripts/fuseki-repository-config.ttl.template   stays as is
> knora-ontologies/xyz.ttl                               webapi/src/main/resources/knora-ontologies/xyz.ttl
> test_data/project_data/xyz.ttl                         stays as is
> test_data/project_ontologies/xyz.ttl                   stays as is
> ```

Looks right, but I would suggest quickly testing it to verify, once this is merged, just to be sure.

**Pull request merged**